### DISCUSSION TITLE
Feature Request: On demand inline suggestions

### DISCUSSION THREAD

**Discussion by drew-y** [community/community#9817] [Copilot Conversations]
By default, co-pilot automatically makes suggestions inline as your write code. This behavior can get annoying in many cases. It would be great if instead we could configure copilot make an inline suggestion after pressing a keyboard shortcut.

I'm aware of the `ctrl ^` shortcut. But this opens suggestions in a new page, not inline.

Apologize if this is already possible / suggested (if so links would be helpful).

**Comment by drdaxxy** [comment]
There *is* a *Trigger Inline Suggestion* command, bound by default to <code><i>Alt</i>+^</code>... but it doesn't work if automatic suggestions are disabled :(

Although the default keybinding has `config.github.copilot.inlineSuggest.enable` in its trigger condition, even if we remove that, the command itself does nothing.

This is with current (v1.7.4459) nightly in VS Code.

**Comment by bobmagicii** [comment]
yep. still a problem.

**Comment by m-gris** [comment]
Any update / progress on that matter ? :-)

**Comment by ChuckJonas** [comment]
Please add this feature.  In many context Co-Pilot gets in the way and is a distraction.  I would prefer to be able to ask for completions only when I feel they are appropriate

**Comment by bitbytebit1** [comment]
We need traction on this issue so badly

**Comment by MartinodF** [comment]
Still an issue. It's really unfortunate VSCode users can't rely on on-demand inline suggestions but have to pick between always-on automatic inline suggestions or no inline suggestions at all.

**Comment by SmallLemur** [comment]
We shipped this feature a few days ago now :)
The Copilot version that has the fix is 1.84.51. There is a new setting in VSCode (1.77) that can be used to disable the auto complete. Once disabled you can trigger a completion using ALT/OPTION + \ like in IntelliJ. Please note that disabling inline suggestion will disable both, manual and auto completions.
<img width="1255" alt="image" src="https://user-images.githubusercontent.com/16976466/234612498-8b42212d-9737-4720-8158-e72dc6ec4da6.png">

**Reply by 4722794** [reply]
Thank you petiteLemur!
This was such a wonderful addition.
Copilot is the best AI product out there yet!

**Reply by michaelkhuber** [reply]
@SmallLemur in the meantime, the settings changed and this is now deprecated. Any other way to get this behaviour now?

**Reply by hamzace** [reply]
I could make it work by:
1. Adding `"editor.inlineSuggest.enabled": false,` in `settings.json`
2. Adding to `keybindings.json`
```
  {
    "key": "<YOUR KEYBIND>",
    "command": "editor.action.inlineSuggest.trigger",
    "when": "editorTextFocus && !inlineSuggestionVisible"
  }
```

@michaelkhuber Hope that helps :)

**Reply by S0yKaf** [reply]
@hamzace are there any side effects to setting inlineSuggest.enabled to false? It's honestly quite annoying that Microsoft keeps breaking this toggle every couple months.

Copilot is super useful. but really I want to decide when it runs.

**Reply by hamzace** [reply]
@S0yKaf I didn't experience any side effect on my end. IntelliSense is still working fine, and I trigger copilot suggestions only when needed (and honestly that doesn't happen often)

**Reply by pumpkinlink** [reply]
the current setting is global to the whole vscode, not just copilot: `editor.inlineSuggest.enabled` , there's an experimental one to disable only some extensions (`editor.inlineSuggest.experimental.suppressInlineSuggestions`), but this one didn't work for me, only the global one.

**Reply by mfp22** [reply]
I had this set up in the past and can't figure out how to get it back. The settings got more complicated and I have a hard time believing that people actually like being rapidly and unpredictably interrupted with suggestions that shift content and alter the behavior of the tab key.

**Comment by GARJE-01** [comment]
If you previously had "manual-only" Copilot completions working and it stopped after an update, I'd check whether the breakage is coming from VS Code rather than Copilot itself.

A few things to verify:

1. Confirm that your keybinding is still pointing to:
   `editor.action.inlineSuggest.trigger`

2. Open **Developer: Toggle Keyboard Shortcuts Troubleshooting** and press your shortcut. VS Code will show whether another extension or keybinding is intercepting it.

3. Check whether the trigger command works from the Command Palette. If it works there but not from your shortcut, the issue is almost certainly a keybinding conflict.

4. Review any newer AI-related extensions you have installed. Some extensions now contribute inline completions through the same VS Code APIs, which can make it look like Copilot settings are being ignored.

5. If you're using:

   ```json
   "editor.inlineSuggest.enabled": false
   ```

   remember that this is still a global VS Code setting. Depending on the VS Code version, disabling inline suggestions globally can affect all providers, not just Copilot.

I'm curious whether your issue is:

* no suggestion appears when manually triggered,
* automatic suggestions returned unexpectedly, or
* the trigger command no longer fires at all.

Those point to different causes and would help narrow down whether this is a Copilot regression or a VS Code configuration change.

**Reply by mfp22** [reply]
Thank you for the helpful debug steps!

1. Yep
2. Looks good. It logged this: `[KeybindingService]: + Invoking command editor.action.inlineSuggest.trigger`
3. It does not work from the command palette. A way to know what is registered to handle a command would be great.
4. I took a break from development for a while but do not remember installing any additional AI extensions. A way other than manually reviewing all my extensions would be good. If there is, let me know. 
5. No, `editor.inlineSuggest.enabled` is checked as Enabled.

For anyone else struggling with this, while looking into these, I did find `github.copilot.chat.completions.toggle` and temporarily have bound that to `^ + \` and so far it's better than having them permanently off.

**Comment by ChiIIBiII** [comment]
How was this implemented and they broke it, can't believe I searched too long for the setting thinking I broke it myself.
